src/Raven.Studio/typescript/viewmodels/database/query/query.ts

```diff
@@ -757,6 +766,9 @@ class query extends viewModelBase {
             return;
         }
         
+        // if index was defined outside of current view then we don't know that at this point, so must fetch
+        this.fetchAllIndexes(this.activeDatabase());
```

Comment:
wait it means, we *have to* fetch *all* indexes on *every* query?

Reply:
https://issues.hibernatingrhinos.com/issue/RavenDB-15509#focus=Comments-67-49974.0-0